js/main.js

```diff
@@ -20,8 +20,102 @@ function handleSubmit(event) {
   data.nextEntryId++;
   data.entries.unshift(entry);
   $img.setAttribute('src', 'images/placeholder-image-square.jpg');
+  var instantEntry = entryTemplate(data.entries[0]);
+  $ul.prepend(instantEntry);
   $form.reset();
+  switchView($entriesPage.getAttribute('data-view'));
+  data.view = 'entries';
```

Comment:
You should set `data.view` in `switchView` when you find the correct view, so you don't need to remember to do it every time you switch views.